students/tonylee/mailroom.py

```diff
@@ -0,0 +1,48 @@
+donors = ["Tony Lee", "Andy Arko", "Michelle Lee", "Tom Ludwinski", "Kurt Lafser"]
+totalDonationAmount = [100, 200, 300, 400, 500]
+numGifts = [1, 1, 1, 1, 1]
+
+def send_thankYou():
+    donorName = input("Enter the donor's full name: ")
+    #if user types 'list'
+    while donorName == "list":
+        for item in donors:
+            print(item)
+        donorName = input("Enter the donor's full name: ")
+
+    if isSameDonor(donorName) is not True:
+        donors.append(donorName)
+        totalDonationAmount.append(0.0)
+        numGifts.append(0)
+        setDonation(len(donors)-1, getDonation())
+
+    print(f'Thank you {donorName}, for your donation!')
+
+def create_report():
+    print('{:20}|{:15}|{:12}|{}'.format('Donor Name', 'Total Given', 'Num Gifts', 'Average Gift'))
+    for i in range(len(donors)):
+        print('{:20} ${:10,.2f}{:^17} ${:10,.2f}'.format(donors[i], totalDonationAmount[i], numGifts[i], totalDonationAmount[i]/numGifts[i]))
+
+def isSameDonor(donorName):
+    for i in range(len(donors)):
+        if donors[i] == donorName:
+            setDonation(i, getDonation())
```

Comment:
this should be done outside of this function - it is best practice for functions to have side-effects, meaning:

if function implies that it returns a boolean then it should be a function that only does; here it is doing a boolean check + modifies that data.